### ISSUE TITLE
io_uring test "accept/connect/recv/cancel" fails on Linux pc 5.17.1-arch1-1

### ISSUE THREAD

**Issue by matu3ba** [ziglang/zig#11382]
### Zig Version

0.10.0-dev.1717+91eb1af91

### Steps to Reproduce

1. zig-bootstrap:
- `./build -j2 x86_64-linux-musl native`
2. build Zig:
- `cd build/ && cmake .. -DZIG_STATIC_LLVM=on -DCMAKE_PREFIX_PATH="$HOME/dev/git/bootstrap/zig-bootstrap/musl/out/host/" -GNinja && /usr/bin/time -v ${HOME}/dev/git/cpp/mold/mold -run ninja && cd ..`
3. run `zig test lib/std/os/linux/io_uring.zig` or `./build/zig test lib/std/os/linux/io_uring.zig`

Sidenode: The testname with forward slashes is annoying to filter.

### Expected Behavior

test "accept/connect/recv/cancel" passes

### Actual Behavior

test "accept/connect/recv/cancel" does not pass with:

```txt
./build/zig test lib/std/os/linux/io_uring.zigTest [17/24] test "accept/connect/recv/cancel"... expected 4294967295, found 2576980377
Test [17/24] test "accept/connect/recv/cancel"... FAIL (TestExpectedEqual)
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:39:27: 0x22a4bf in IO_Uring.init_params (test)
        if (entries == 0) return error.EntriesZero;
                          ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:31:16: 0x22472d in IO_Uring.init (test)
        return try IO_Uring.init_params(entries, &params);
               ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:40:46: 0x22a506 in IO_Uring.init_params (test)
        if (!std.math.isPowerOfTwo(entries)) return error.EntriesNotPowerOfTwo;
                                             ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:31:16: 0x22472d in IO_Uring.init (test)
        return try IO_Uring.init_params(entries, &params);
               ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:143:46: 0x22c2f9 in IO_Uring.get_sqe (test)
        if (next -% head > self.sq.sqes.len) return error.SubmissionQueueFull;
                                             ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:355:21: 0x224a18 in IO_Uring.nop (test)
        const sqe = try self.get_sqe();
                    ^
/home/user/dev/git/zig/zig/master/lib/std/testing.zig:79:17: 0x226055 in std.testing.expectEqual (test)
                return error.TestExpectedEqual;
                ^
/home/user/dev/git/zig/zig/master/lib/std/testing.zig:120:17: 0x225325 in std.testing.expectEqual (test)
                try expectEqual(@field(expected, field.name), @field(actual, field.name));
                ^
/home/user/dev/git/zig/zig/master/lib/std/os/linux/io_uring.zig:2405:5: 0x21e9c0 in test "accept/connect/recv/cancel" (test)
    try testing.expectEqual(linux.io_uring_cqe{
    ^
23 passed; 0 skipped; 1 failed.
error: the following test command failed with exit code 1:
lib/std/os/linux/zig-cache/o/c72b9b73ab99d9108a2733069ef73467/test /home/user/dev/git/zig/zig/master/build/zig
```

**Comment by briangold** [comment]
I reproduced this using Arch Linux 5.17.5-arch1-1, although I suspect we might see it on any bleeding edge kernel. 

The issue: the test in question submits a `recv` and a subsequent `cancel` operation to io_uring. The cancel causes both operations to complete, generating entries in the io_uring completion queue. The test was assuming the recv completion would come before the cancel completion, but io_uring doesn't make any such guarantees, and we were seeing a reverse order of completions.

I added a check for this with appropriate swap operation, similar to what is done in the same test for handling "racing" accept and connect operations. 

The tests all pass for me on both this ArchLinux environment and in my normal NixOS environment.
